src/views/Activate/ActivateCode.js

```diff
@@ -1,157 +1,152 @@
-    autoFocus: true,
-  });
-
   const goToLogin = useCallback(() => history.popAndPush(ROUTE_NAMES.LOGIN), [
     history,
   ]);
+
   const goToPassport = useCallback(() => {
     push(names.PASSPORT);
+
     if (!hasDisclaimed) {
       push(names.DISCLAIMER);
     }
   }, [names, push, hasDisclaimed]);
 
-  const pass = derivedWallet.matchWith({
-    Nothing: () => false,
-    Just: () => true,
-  });
+  const validate = useMemo(
+    () => composeValidator({ ticket: buildPatqValidator() }),
+    []
+  );
 
-  useEffect(() => {
-    if (validTicket) {
-      const _contracts = need.contracts(contracts);
+  // set our state on submission
+  const onSubmit = useCallback(
+    async values => {
+      await timeout(100); // allow the ui changes to flush before we lag it out
 
-      setDeriving(true);
-      // when the ticket becomes valid, derive the point
-      (async () => {
-        const { seed } = await generateTemporaryOwnershipWallet(ticket);
-
-        //TODO isn't all this accessible in the ownership object?
-        const inviteWallet = walletFromMnemonic(seed, DEFAULT_HD_PATH);
-        const _inviteWallet = need.wallet(inviteWallet);
-
-        const owned = await azimuth.azimuth.getOwnedPoints(
-          _contracts,
-          _inviteWallet.address
-        );
-        const transferring = await azimuth.azimuth.getTransferringFor(
-          _contracts,
-          _inviteWallet.address
-        );
-        const incoming = [...owned, ...transferring];
-
-        let realPoint = Nothing();
-        let wallet = Nothing();
-
-        if (incoming.length > 0) {
-          const pointNum = parseInt(incoming[0], 10);
-          realPoint = Just(pointNum);
-          wallet = Just(await generateWallet(pointNum));
-
-          if (incoming.length > 1) {
-            setGeneralError(
-              'This invite code has multiple points available.\n' +
-                "Once you've activated this point, activate the next with the same process."
-            );
-          } else {
-            setGeneralError(false);
-          }
-        } else {
-          setGeneralError(
-            'Invite code has no claimable point.\n' +
-              'Check your invite code and try again?'
-          );
+      const _contracts = need.contracts(contracts);
+      const { seed } = await generateTemporaryOwnershipWallet(values.ticket);
+
+      const inviteWallet = walletFromMnemonic(seed, DEFAULT_HD_PATH);
+
+      const _inviteWallet = need.wallet(inviteWallet);
+
+      const owned = await azimuth.azimuth.getOwnedPoints(
+        _contracts,
+        _inviteWallet.address
+      );
+      const transferring = await azimuth.azimuth.getTransferringFor(
+        _contracts,
+        _inviteWallet.address
+      );
+      const incoming = [...owned, ...transferring];
+
+      if (incoming.length > 0) {
+        if (incoming.length > 1) {
+          // TODO: putting a warning here doesn't make sense since the user
```

Comment:
I moved the validation here to onSubmit rather than in realtime, so now this warning doesn't make sense to display — what do?